src/Lean/Compiler/InitAttr.lean

```diff
@@ -70,7 +71,11 @@ unsafe def registerInitAttrUnsafe (attrName : Name) (runAfterImport : Bool) : IO
   }
 
 @[implementedBy registerInitAttrUnsafe]
-opaque registerInitAttr (attrName : Name) (runAfterImport : Bool) : IO (ParametricAttribute Name)
+private opaque registerInitAttrInner (attrName : Name) (runAfterImport : Bool) (ref : Name) : IO (ParametricAttribute Name)
```

Comment:
This is needed because `implementedBy` does not seem to like auto params; I tried putting the auto param on both or just one or the other and they all get rejected. So I settled for another wrapper around the unsafe impl. (We should probably move term-`unsafe` to lean core to clean this kind of thing up.)

Reply:
Right, `implementedBy` currently uses structural equality, and the two tactics are not equal anymore after extraction into separate declarations.

> We should probably move term-unsafe to lean core to clean this kind of thing up.

Yes, I think that would be a good idea.